form-test-tom/src/useForm.js

```diff
@@ -0,0 +1,78 @@
+import {useState} from 'react'
+
```

Comment:
this custom react hook is the result of a tutorial and applying the lessons therein to this problem space. 
the approach was appealing: a custom hook for form validation which is reusable (thanks to validation options and other config passed to the function), extensible (there are other forms of validation that can be quite esoteric. Captchas, anyone?), and modern in flavour.